### DISCUSSION TITLE
Custom compiling for a specific old non-AVX-CPU

### DISCUSSION THREAD

**Discussion by rmast** [Xaymar/obs-stream-effects#965] [Q&A] [answered]
In the system requirements I read AVX is mandatory. However in the CMakelists I see an option for native building, however underneath 'NOT MSVC'. 

My quadcore AMD Athlon(tm) II X4 645 Processor publishes these flags in /proc/cpuinfo:
flags		: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm 3dnowext 3dnow constant_tsc rep_good nopl nonstop_tsc cpuid extd_apicid pni monitor cx16 popcnt lahf_lm cmp_legacy svm extapic cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw ibs skinit wdt hw_pstate vmmcall npt lbrv svm_lock nrip_save

What would be the best direction of my attempt to get it compiled and running? Only keep TARGET_X86_64 on and build in MSVC in Windows, or go for Linux/gcc and build native?

By the way, the PC has a nvidia 1030 GT GPU and my aim is to get a watermark-free webcam-blur-background for Teams.

**Comment by Xaymar** [comment, marked as answer]
StreamFX supports native building on all three supported platforms, with varyous compilers. Most if not all StreamFX code is standard C++17 or older, and as such should work with recent versions of MSVC, GCC, Clang and AppleClang.

As for the TARGET_X86_64 variables, take a look [on wikipedia](https://en.wikipedia.org/wiki/X86-64#Microarchitecture_levels) for what they mean, or use `TARGET_NATIVE` on Unix platforms to build for the current CPU only. Building for native means that the builds can't be shared to other people, but they may make far better use of the current CPU. If you don't plan to share the builds, building for native is always recommended for improved performance.
